pkg/service/v0/service.go

```diff
@@ -23,30 +27,84 @@ func NewService(opts ...Option) Service {
 	svc := Thumbnails{
 		config: options.Config,
 		mux:    m,
+		manager: thumbnails.SimpleManager{
+			Storage: storage.NewFileSystemStorage(options.Config.FileSystemStorage),
+		},
+		source: imgsource.NewWebDavSource(options.Config.WebDavSource),
 	}
 
 	m.Route(options.Config.HTTP.Root, func(r chi.Router) {
-		r.Get("/", svc.Dummy)
+		r.Get("/thumbnails", svc.Thumbnails)
 	})
 
 	return svc
 }
 
 // Thumbnails defines implements the business logic for Service.
 type Thumbnails struct {
-	config *config.Config
-	mux    *chi.Mux
+	config  *config.Config
+	mux     *chi.Mux
+	manager thumbnails.Manager
+	source  imgsource.Source
 }
 
 // ServeHTTP implements the Service interface.
 func (g Thumbnails) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	g.mux.ServeHTTP(w, r)
 }
 
-// Dummy implements the Service interface.
-func (g Thumbnails) Dummy(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Content-Type", "text/plain")
-	w.WriteHeader(http.StatusOK)
+// Thumbnails provides the endpoint to retrieve a thumbnail for an image
+func (g Thumbnails) Thumbnails(w http.ResponseWriter, r *http.Request) {
+	query := r.URL.Query()
+	width, _ := strconv.Atoi(query.Get("width"))
+	height, _ := strconv.Atoi(query.Get("height"))
+	fileType := query.Get("type")
+	filePath := query.Get("file_path")
+	etag := query.Get("etag")
 
-	w.Write([]byte("Hello ocis-thumbnails!"))
+	encoder := thumbnails.EncoderForType(fileType)
+	if encoder == nil {
+		// TODO: better error responses
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte("can't encode that"))
```

Comment:
use a logger from `ocis-pkg` so we have persisted logs and not only the user can see it. I'm also fond of using the passive voice, as in:

> can't be encoded

Furthermore the error message is not informative enough, since you're checking whether the `fileType` is supported or not by fetching an encoder for it, it should read more like:

> encoding for filetype: .XYZ is not supported

Reply:
I wanted to replace that when the proper API is implemented.